integration_tests/src/main/python/delta_lake_time_travel_test.py

```diff
@@ -0,0 +1,139 @@
+# limitations under the License.
+
+import datetime
+
+
+from asserts import *
+from data_gen import *
+from delta_lake_utils import delta_meta_allow, enable_in_commit_ts
+from marks import *
+from spark_session import with_gpu_session
+from delta.tables import *
+
+
+in_commit_ts_param_id = lambda val: f"in_commit_ts={val}"
+
+@delta_lake
+def test_time_travel_on_non_existing_table():
+    def time_travel_on_non_existing_table():
+        with_gpu_session(lambda spark: spark.sql("SELECT * FROM not_existing VERSION AS OF 0"))
+
+    assert_spark_exception(time_travel_on_non_existing_table, "AnalysisException")
+
+def do_set_up_tables_for_time_travel(spark_tmp_path, spark_tmp_table_factory, in_commit_ts = True,
+                                     times = 2):
+    table = spark_tmp_table_factory.get()
+    table_path = f"{spark_tmp_path}/{table}"
+
+    def setup_delta_table(spark):
+        spark.sql(f"CREATE TABLE {table}(a int, b string) USING delta LOCATION '{table_path}'")
+        if in_commit_ts:
+            spark.sql(f"ALTER TABLE {table} SET TBLPROPERTIES ('delta.enableInCommitTimestamps' = 'true');")
+        df = two_col_df(spark, int_gen, string_gen)
+        df.write.mode("append").format("delta").save(table_path)
+
+
+    def append_to_delta_table(spark):
+        df = two_col_df(spark, int_gen, string_gen)
+        df.write.mode("append").format("delta").save(table_path)
+
+    with_cpu_session(setup_delta_table)
+
+    if times > 1:
+        for _ in range(times - 1):
+            with_cpu_session(append_to_delta_table)
+
+    return table_path
+
+def do_get_delta_table_timestamps(spark, table_path) -> Dict[int, datetime]:
+    delta_table = DeltaTable.forPath(spark, table_path)
+    commit_history_rows = delta_table.history().select("version", "timestamp").collect()
+    def convert_ts(ts):
+        if isinstance(ts, (int, float)):
+            # timestamp is in microseconds
+            return datetime.fromtimestamp(ts / 1000_000.0)
+        else:
+            return ts
+    commit_map = {row.version: convert_ts(row.timestamp) for row in commit_history_rows}
+    return commit_map
+
+
+@allow_non_gpu(*delta_meta_allow)
+@delta_lake
+@ignore_order(local=True)
+@pytest.mark.parametrize("in_commit_ts", enable_in_commit_ts(), ids=in_commit_ts_param_id)
+def test_time_travel_df_version(spark_tmp_path, spark_tmp_table_factory, in_commit_ts):
+    table_path = do_set_up_tables_for_time_travel(spark_tmp_path, spark_tmp_table_factory,
+                                                  in_commit_ts,
+                                                  times = 3)
+
+    def check_version(spark, version):
+        return spark.read.format("delta").option("versionAsOf", version).load(table_path)
+
+    assert_gpu_and_cpu_are_equal_collect(lambda spark: check_version(spark, 0))
+    assert_gpu_and_cpu_are_equal_collect(lambda spark: check_version(spark, 1))
+    assert_gpu_and_cpu_are_equal_collect(lambda spark: check_version(spark, 2))
```

Comment:
Should `check_version` always return a non-empty dataframe? If so, please add a check for that as well. We should make sure that it returns a valid result.

Reply:
Same comment for other tests.